pkg/apis/core/validation/shoot_test.go

```diff
@@ -538,6 +543,18 @@ var _ = Describe("Shoot Validation Tests", func() {
 			)
 		})
 
+		It("should allow updating the seed if it has been set previously and the SeedChange feature gate is enabled", func() {
+			defer test.WithFeatureGate(utilfeature.DefaultFeatureGate, features.SeedChange, true)()
```

Comment:
Can someone explain why this is done in the defer?

Reply:
This is similar to the `WithVars` and other methods that we already have in the `utils/test` package and called in a similar way in all tests that make use of it.

Reply:
@danielfoehrKn Yes, you understand it correctly. This is a "pattern" in Go, and `defer` is often used exactly in this way. The idea is that you achieve in a single line, and with a single function what you would otherwise achieve with 2 lines and 2 functions, and all passing of parameters that you would otherwise have to do yourself (e.g. the of the original value) happens automatically. Other functions in the `test` package also follow this pattern. In Kubernetes itself, there is a function `featuregatetesting.SetFeatureGateDuringTest` that also follows this pattern, unfortunately it can't be used directly with Ginkgo since it requires access to `testing.T`, so I basically re-implemented it here for Ginkgo.